src/modules/Form/index.tsx

```diff
@@ -0,0 +1,234 @@
+import { FIELDS, FIELDS_VALIDATION_BY_NAME, ERROR_MESSAGES, FEMALE, MALE } from 'appConstants';
+import { CommonFieldType, CustomPokemon, GenderFieldType, SelectFieldType } from 'types';
+import React, { createRef, FormEvent, RefObject } from 'react';
+import { Layout } from 'modules';
+import * as S from './styled';
+import { uuid } from 'utils';
+import {
+  BirthdayField,
+  ErrorMessage,
+  ConsentField,
+  GenderField,
+  AvatarField,
+  ShinyField,
+  TypeField,
+  NameField,
+  FormCard,
+} from './components';
+
+type Props = {
+  componentName: string;
+  location: string;
+};
+
+type State = {
+  pokemons: Array<CustomPokemon & { id: string }>;
+  errorFields: Array<string>;
+  initialEnter: boolean;
+};
+
+export class Form extends React.Component<Props, State> {
+  isShinyField: Pick<CommonFieldType, 'inputRef'>;
+  successMessage: RefObject<HTMLSpanElement>;
+  birthdayField: CommonFieldType;
+  consentField: CommonFieldType;
+  genderField: GenderFieldType;
+  avatarField: CommonFieldType;
+  typeField: SelectFieldType;
+  nameField: CommonFieldType;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      pokemons: [],
+      errorFields: [],
+      initialEnter: true,
+    };
+    this.successMessage = createRef();
+    this.birthdayField = {
+      inputRef: createRef(),
+      errorRef: createRef(),
+    };
+    this.consentField = {
+      inputRef: createRef(),
+      errorRef: createRef(),
+    };
+    this.isShinyField = {
+      inputRef: createRef(),
+    };
+    this.genderField = {
+      femaleInputRef: createRef(),
+      maleInputRef: createRef(),
+      errorRef: createRef(),
+    };
+    this.avatarField = {
+      inputRef: createRef(),
+      errorRef: createRef(),
+    };
+    this.typeField = {
+      selectRef: createRef(),
+      errorRef: createRef(),
+    };
+    this.nameField = {
+      inputRef: createRef(),
+      errorRef: createRef(),
+    };
+  }
+
+  removeIncorrectField = (field: string) => {
+    if (this.state.initialEnter) this.setState({ initialEnter: false });
+    const fields = [...this.state.errorFields];
+    const index = fields.indexOf(field);
+    if (index !== -1) {
+      fields.splice(index, 1);
+      this.setState({ errorFields: [...fields] });
```

Comment:
```
const { name, ...rest} = this.state.errors;
this.setState({ errors: rest });
```
_________
this.setState({ errors: {...errors, name: null} });

_____
this.setState({ errors: {...errors, name: {errorMsg: ''} });